src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -424,7 +435,32 @@ void WCSimPrimaryGeneratorAction::GeneratePrimaries(G4Event* anEvent)
       particleGun->GeneratePrimaryVertex(anEvent);
     }//old muline format
   }//useMuLineEvt
+  
+  else if(useAmBeEvt){ // Diego Costas (diego.costas.rodriguez@usc.es) 2023
+    // Initialise the ambe generator once per sim
+    // This will get AmBe settings (position, direction, etc)
+    if ( !AmBeGen ){
+      AmBeGen = new WCSimAmBeGen();
+    }
+
+    generateBGOEvent = IsBGOEventSet(); 
```

Comment:
I don't think you need the `generateBGOEvent` variable and relevant functions.

Reply:
I wrote this so this is the way I implemented the logic behind `/mygen/BGOEvent`. Is it not correct, then?